internal/backend/remote-state/http/client_test.go

```diff
@@ -251,3 +253,93 @@ func TestHttpClient_IsLockingEnabled(t *testing.T) {
 		})
 	}
 }
+
+// Tests the Lock method for the HTTP client.
+// Test to see correct lock info is returned
+func TestHttpClient_lock(t *testing.T) {
+	stateLockInfoA := statemgr.LockInfo{
+		ID:        "ada-lovelace-state-lock-id",
+		Who:       "AdaLovelace",
+		Operation: "TestTypePlan",
+		Created:   time.Date(2023, time.August, 16, 15, 9, 26, 0, time.UTC),
+	}
+
+	testCases := []struct {
+		name                string
+		lockInfo            *statemgr.LockInfo
+		lockResponseStatus  int
+		lockResponseBody    []byte
+		expectedStateLockID string
+		expectedErrorMsg    error
+	}{
+		{
+			// Successful locking HTTP remote state
+			name:                "Successfully locked",
+			lockInfo:            &stateLockInfoA,
+			lockResponseStatus:  http.StatusOK,
+			lockResponseBody:    nil,
+			expectedStateLockID: stateLockInfoA.ID,
+			expectedErrorMsg:    nil,
+		},
+		{
+			// Failed to lock state, HTTP remote state already locked
+			name:                "Locked remote state",
+			lockInfo:            &stateLockInfoA,
+			lockResponseStatus:  http.StatusLocked,
+			lockResponseBody:    []byte(`{"ID":"linus-torvalds-http-remote-state-lock-id", "Path": "", "Operation": "TestTypePlan", "Who": "LinusTorvalds", "Created": "2024-08-15T09:00:26.00Z" }`),
+			expectedStateLockID: "",
+			expectedErrorMsg: &statemgr.LockError{
+				Info: &statemgr.LockInfo{
+					ID:        "linus-torvalds-http-remote-state-lock-id",
+					Who:       "LinusTorvalds",
+					Operation: "TestTypePlan",
+					Created:   time.Date(2024, time.August, 15, 9, 0, 26, 0, time.UTC),
+				},
```

Comment:
nit (could be ignored): we can move this as a separate variable and reuse it for json generation and in the error message.

Reply:
thanks for the feedback, I have already implemented it, I haven't pushed that change yet.

Reply:
Moved it to it's own variable.